source/numerics/matrix_creator.inst.in

```diff
@@ -67,42 +67,42 @@ for (scalar: REAL_AND_COMPLEX_SCALARS; deal_II_dimension : DIMENSIONS; deal_II_s
     (const Mapping<deal_II_dimension,deal_II_space_dimension>       &mapping,
      const DoFHandler<deal_II_dimension,deal_II_space_dimension>    &dof,
      const Quadrature<deal_II_dimension>    &q,
-     SparseMatrix<numbers::NumberTraits<scalar>::real_type>     &matrix,
+     SparseMatrix<scalar>     &matrix,
      const Function<deal_II_space_dimension,scalar>      &rhs,
      Vector<scalar>           &rhs_vector,
-     const Function<deal_II_space_dimension,numbers::NumberTraits<scalar>::real_type> * const coefficient,
-     const ConstraintMatrix   &constraints);
+     const Function<deal_II_space_dimension,scalar> * const coefficient,
+     const AffineConstraints<scalar>   &constraints);
     template
     void MatrixCreator::create_mass_matrix<deal_II_dimension,deal_II_space_dimension,scalar>
     (const DoFHandler<deal_II_dimension,deal_II_space_dimension>    &dof,
      const Quadrature<deal_II_dimension>    &q,
-     SparseMatrix<numbers::NumberTraits<scalar>::real_type>     &matrix,
+     SparseMatrix<scalar>     &matrix,
      const Function<deal_II_space_dimension,scalar>      &rhs,
      Vector<scalar>           &rhs_vector,
-     const Function<deal_II_space_dimension,numbers::NumberTraits<scalar>::real_type> * const coefficient,
-     const ConstraintMatrix   &constraints);
+     const Function<deal_II_space_dimension,scalar> * const coefficient,
+     const AffineConstraints<scalar>   &constraints);
```

Comment:
@bangerth This is another intrusive change I made: I unified the interface for `create_mass_matrix` and `create_boundray_mass_matrix` to uniformly require the same number type. I think this is the right approach and the previous version that required complex-valued vectors but real-valued matrices is a cludge. First, we should really support mass matrices with a complex-valued coefficient because this shows up frequently in applications (think about dualized time-harmonic Maxwell's equation / complex-valued Helmholtz equation with a complex-valued permeability/permittiivity).

The only open problem with that is that ...